components/src/composables/useFlashing.ts

```diff
@@ -0,0 +1,38 @@
+import {ShallowReactive, ref, watch} from 'vue';
+
+export default function useFlashing(
+  props: ShallowReactive<{
+    order: object;
+    items: object[];
+    loading: boolean;
+  }>,
+) {
+  let oldValue: object[] = [];
+  const flashIndexes = ref(Array(props.items.length).fill(false));
+
+  // Reset cached values if loading or sorting
+  watch([() => props.loading, () => props.order], () => (oldValue = []), {
+    flush: 'pre',
+  });
+
+  watch(
+    () => props.items,
+    newValue => {
+      flashIndexes.value = Array(newValue.length).fill(false);
+      for (let i = 0; i < flashIndexes.value.length; i++) {
```

Comment:
O(n)^2 because having to iterate on both arrays. :cry:

Reply:
I also thought about this. but hashing itself takes some complexity. I performance tested a very simple hash function and in best case it takes about 1.5ms per hash

https://gist.github.com/hyamamoto/fd435505d29ebfa3d9716fd2be8d42f0?permalink_comment_id=4557681#gistcomment-4557681